engine/crates/parser-sdl/src/rules/mongodb_directive/type_directive.rs

```diff
@@ -16,15 +16,24 @@ impl<'a> Visitor<'a> for MongoDBTypeDirective {
             return;
         }
 
-        let TypeKind::Object(ref object) = r#type.node.kind else {
-            return;
-        };
+        let type_name = r#type.name.as_str();
 
-        let input_type_name = generic::filter_type_name(r#type.name.as_str());
-        filter::register_type_input(ctx, object, &input_type_name, std::iter::empty());
-        generic::register_array_type(ctx, r#type.name.as_str(), false);
-        generic::register_singular_type(ctx, r#type.name.as_str());
-        filter::register_orderby_input(ctx, object, r#type.node.name.as_str(), std::iter::empty());
-        input::register_type_input(ctx, object, &r#type.node);
+        match &r#type.node.kind {
+            TypeKind::Object(object) => {
+                let input_type_name = generic::filter_type_name(type_name);
+                filter::register_type_input(ctx, object, &input_type_name, std::iter::empty());
+                generic::register_array_type(ctx, r#type.name.as_str(), false);
+                generic::register_singular_type(ctx, r#type.name.as_str());
+                filter::register_orderby_input(ctx, object, r#type.node.name.as_str(), std::iter::empty());
+                input::register_type_input(ctx, object, &r#type.node);
+            }
+            TypeKind::Enum(_enum) => {
```

Comment:
We generate these input types for objects (right above), as well as for scalars, but not for enums. That makes the GraphQL schema of the engine invalid, and that seems like an omission.